### PR TITLE
Added Christmas Giftgiving Goal

### PR DESCRIPTION
**Pull request by DaveKorhal**
Added a Christmas goal to give a random crewmate a gift.

## What Does This PR Do
During Christmas, there's a chance for the Christmas greeting to encourage the crew to give a gift to a random crewmate.

## Why It's Good For The Game
Adds a role-play objective during the Christmas season.

## Changelog
:cl:
add: New random objective during Christmas
/:cl:

### CODE REVIEW

File: code/modules/holiday/holiday.dm [hunk 1]
```diff
@@ -232,7 +232,10 @@
 	eventChance = 20
 
 /datum/holiday/xmas/greet()
-	return "Have a merry Christmas!"
+	var/Greetings = "Have a merry Christmas!"
+	if(prob(30))
+		Greetings += "<br><br>To celebrate, choose a random crewmate on the Manifest and give them a gift!"
+	return Greetings
```
Comment (AffectedArc07) [inline code/modules/holiday/holiday.dm:238]:
```suggestion
	var/greeting = "Have a merry Christmas!"
	if(prob(30))
		greeting += "<br><br>To celebrate, choose a random crewmate on the Manifest and give them a gift!"
	return greeting
```
Casing, and this is a string, not a list, so pluralising it is a bit confusing.

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]

**Comment by kenken244** [comment]
I feel like it would be more interesting if the game picked a few people at random instead of telling them to pick it. Otherwise they would just give things to their meta friends

**Comment by Spacemanspark** [comment]
They're going to do that anyways tho

**Comment by kenken244** [comment]
Then this way they give gifts to their friends and also some other people. I see no problem with this

**Comment by DaveKorhal** [comment]
> I feel like it would be more interesting if the game picked a few people at random instead of telling them to pick it. Otherwise they would just give things to their meta friends

That was my original plan, but all of the Holiday subsystem code except the Greet() processes before the roundstart crew is even chosen. And the Greet() is a global announcement so I couldn't give players specific recipients in it either. Since every workaround I've thought of involves messing with either the initialization code or the admin code, I decided to just KISS.

I would _love_ to code a more robust Holiday subsystem to handle that stuff, but that's not how I want to spend Xmas week.

**Comment by DaveKorhal** [comment]
...I suppose I could also put in some timer that either waits X minutes or continuously checks to see if the round started- there's already something like that in the Xmas code- and then choose recipients at random, but when I linked that to SteelSlayer he let out a wail of agony. So I'm guessing the Maintainers probably wouldn't like something that... hacky.

**Review by AffectedArc07 — APPROVED** [review]
May as well, even if it wont be used for another 11 months

**Pull request merged**